src/service/contractService.js

```diff
@@ -34,83 +33,91 @@ class ContractService {
     }
     return this[contractName];
   }
-  getContractAddressesFromHubContract(contractName) {
-    const hubContract = new this.web3.eth.Contract(
-      hubContractAbi,
+  async getContractAddressesFromHubContract(contractName) {
+    const hubContract = new ethers.Contract(
       store.getters.selectedNetwork.hubContract,
+      hubContractAbi,
+      this.ethersSigner,
     );
-    return hubContract.methods.getContractAddress(contractName).call();
+    return await hubContract.getContractAddress(contractName);
   }
 
   async getIdentity(operationalAddress, adminWalletAddress) {
     const address = await this.getContractAddress('IdentityStorage');
-    const IdentityStorageProfile = new this.web3.eth.Contract(IdentityStorage, address);
-    const identity = await IdentityStorageProfile.methods
-      .getIdentityId(operationalAddress)
-      .call({ from: this.web3.eth.defaultAccount });
-    const adminKey = ethers.utils.keccak256(
-      ethers.utils.solidityPack(['address'], [adminWalletAddress]),
+    const IdentityStorageContract = new ethers.Contract(
+      address,
+      IdentityStorage,
+      this.ethersSigner,
     );
-    const hasPurpose = await IdentityStorageProfile.methods
-      .keyHasPurpose(identity, adminKey, 1)
-      .call({ from: this.web3.eth.defaultAccount });
+    const identity = await IdentityStorageContract.getIdentityId(operationalAddress);
+    const adminKey = this.getAdminKeyFromWallet(adminWalletAddress);
+    const hasPurpose = await IdentityStorageContract.keyHasPurpose(identity, adminKey, 1);
```

Comment:
I guess we can make some constant ``ADMIN_KEY_PURPOSE = 1`` to make it more clear

Reply:
constant added